gson/src/main/java/com/google/gson/DefaultDateTypeAdapter.java

```diff
@@ -39,6 +39,7 @@
   private final DateFormat enUsFormat;
   private final DateFormat localFormat;
   private final DateFormat iso8601Format;
+  private final DateFormat iso8601FormatMilli;
```

Comment:
yes, i'll keep one DateFormat object and reduce precision of millisecond to fit ISO8601. Close request